shared/pkg/scanner/trivy/local.go

```diff
@@ -0,0 +1,256 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trivy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	trivyDBTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/commands/artifact"
+	trivyFlag "github.com/aquasecurity/trivy/pkg/flag"
+	trivyTypes "github.com/aquasecurity/trivy/pkg/types"
+
+	"github.com/openclarity/kubeclarity/shared/pkg/config"
+	"github.com/openclarity/kubeclarity/shared/pkg/job_manager"
+	"github.com/openclarity/kubeclarity/shared/pkg/scanner"
+	"github.com/openclarity/kubeclarity/shared/pkg/utils"
+)
+
+type LocalScanner struct {
+	logger     *log.Entry
+	config     config.LocalScannerTrivyConfigEx
+	resultChan chan job_manager.Result
+	localImage bool
+}
+
+func (a *LocalScanner) Run(sourceType utils.SourceType, userInput string) error {
+	a.logger.Infof("Called %s scanner on source %v %v", ScannerName, sourceType, userInput)
+	go func() {
+		var output bytes.Buffer
+
+		dbRepoDefaultValue, ok := trivyFlag.DBRepositoryFlag.Value.(string)
+		if !ok {
+			a.setError(fmt.Errorf("unable to get trivy DB repo config"))
+			return
+		}
+
+		severities := []trivyDBTypes.Severity{}
+		for _, name := range trivyDBTypes.SeverityNames {
+			sev, err := trivyDBTypes.NewSeverity(strings.ToUpper(name))
+			if err != nil {
+				a.setError(fmt.Errorf("unable to get trivy severities: %w", err))
+				return
+			}
+			severities = append(severities, sev)
+		}
+
+		trivyOptions := trivyFlag.Options{
+			GlobalOptions: trivyFlag.GlobalOptions{
+				Timeout: a.config.Timeout,
+			},
+			ScanOptions: trivyFlag.ScanOptions{
+				Target: userInput,
+				SecurityChecks: []string{
+					trivyTypes.SecurityCheckVulnerability, // Enable just vuln scanning
+				},
+			},
+			ReportOptions: trivyFlag.ReportOptions{
+				Format:       "json",  // Trivy's own json format is the most complete for vuls
+				ReportFormat: "all",   // Full report not just summary
+				Output:       &output, // Save the output to our local buffer instead of Stdout
+				ListAllPkgs:  false,   // Only include packages with vulnerabilities
+				Severities:   severities,
+			},
+			DBOptions: trivyFlag.DBOptions{
+				DBRepository: dbRepoDefaultValue, // Use the default trivy source for the vuln DB
+				NoProgress:   true,               // Disable the interactive progress bar
+			},
+			VulnerabilityOptions: trivyFlag.VulnerabilityOptions{
+				VulnType: []string{trivyTypes.VulnTypeOS, trivyTypes.VulnTypeLibrary},
+			},
+		}
+
+		var trivySourceType artifact.TargetKind
+		switch sourceType {
+		case utils.IMAGE:
+			trivySourceType = artifact.TargetContainerImage
+		case utils.ROOTFS:
+			trivySourceType = artifact.TargetRootfs
+		case utils.DIR, utils.FILE:
+			trivySourceType = artifact.TargetFilesystem
+		case utils.SBOM:
+			trivySourceType = artifact.TargetSBOM
```

Comment:
looks like it's the same logic in `shared/pkg/analyzer/trivy/trivy.go` - why not to export to a common function?

Reply:
moved into a common function in the latest pushes please take a look.